src/main/java/io/aiven/kafka/auth/json/reader/JsonReader.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) 2020 Aiven, Helsinki, Finland. https://aiven.io/
+ */
+
+package io.aiven.kafka.auth.json.reader;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+
+public interface JsonReader<T> {
+
+    List<T> read() throws JsonReaderException;
+
+    @SuppressWarnings("unchecked")
+    static <K> JsonReader<K> of(final JsonReaderType type, final Path configFile) {
```

Comment:
TBH, I have some trouble seeing a benefit of re-implementing the language's type system over plain constructor calls...

Reply:
agree